Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -959,6 +968,7 @@ private static void CleanupRequest(string requestId)
       printer.Inform("", output); // newline
       printer.Inform($"Verifying {impl.Name} ...", output);
       var verificationResult = GetCachedVerificationResult(impl, output);
+      printer.ReportStartVerifyImplementation(impl);
```

Comment:
Do you also want to report this if a cached result was found? If not, please move this call inside the `if`.

Note that `ReportEndVerifyImplementation` is only called when no cached result is found.

I recommend having the `ReportEndVerifyImplementation` and `ReportStartVerifyImplementation` calls in the same method, so it's easier to see that both are called in the same code path.

Reply:
Yes we want to report this even if a cached result was found. When we invoke Boogie to verify a procedure, it's an internal detail whether it uses the cache or prove it again.
My extension needs the feedback that verification was started and finished for a given implementation, even if there is no split inside.

Good catch, I moved `ReportEndVerifyImplementation ` in the same method as `ReportStartVerifyImplementation `. You actually just discovered a potential bug which did happen to me many times, when my extension reports that something never finishes to verify but Boogie is done.